linux/shim/init.go

```diff
@@ -60,15 +62,27 @@ func newInitProcess(context context.Context, path, namespace string, r *shimapi.
 		}
 		options = *v.(*runcopts.CreateOptions)
 	}
+
+	rootfs := filepath.Join(path, "rootfs")
+	// count the number of successful mounts so we can undo
+	// what was actually done rather than what should have been
+	// done.
+	nrRootMounts := 0
```

Comment:
TL;DR, yes, I should loop until `EINVAL`.

I was thinking it was possible that the caller might have also mounted some stuff on the rootfs before calling create? It'd be weird for `containerd` to undo that. But thinking some more the rootfs here is entirely internal to containerd(-shim) and the user doesn't really formally know where it would be, which I suppose is why were have the rootfs slice in the first place...